pkg/agent/agent_windows.go

```diff
@@ -218,6 +289,74 @@ func (i *Initializer) prepareOVSBridgeOnHNSNetwork() error {
 	return nil
 }
 
+func (i *Initializer) prepareOVSConfigForVM() error {
+	klog.InfoS("Performing OVS configuration", "hostIfName", i.nodeConfig.UplinkNetConfig.Name)
+	hostIfName := i.nodeConfig.UplinkNetConfig.Name
+	uplinkIfName := util.GenerateUplinkInterfaceName(hostIfName)
+	ovsPorts, ovsErr := i.ovsBridgeClient.GetPortList()
+	if ovsErr != nil {
+		klog.Errorf("Failed to list OVS ports")
+		return ovsErr
+	}
+	for index := range ovsPorts {
+		port := &ovsPorts[index]
+		if port.Name == hostIfName {
+			klog.Info("Uplink and host configuration exists in OVS")
+			return nil
+		}
+	}
+
+	success := false
+	uplinkExternalIDs := map[string]interface{}{
+		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaUplink,
+	}
+	// TODO Anand: Have a separate function for creation of pair ports
+	// Create uplink port on OVS.
+	uplinkUUID, ovsErr := i.ovsBridgeClient.CreatePort(uplinkIfName, uplinkIfName, uplinkExternalIDs)
+	if ovsErr != nil {
+		return fmt.Errorf("failed to create uplink port on OVS for %s: %v", uplinkIfName, ovsErr)
+	}
+
+	defer func() {
+		if !success {
+			klog.InfoS("Exiting!! delete port on OVS", "uplinkUUID", uplinkUUID)
+			i.ovsBridgeClient.DeletePort(uplinkUUID)
```

Comment:
Please log the error for these rollback funcs.

Question - what will happen if the rollback fails or agent crashes in the middle? Could agent recover after startup, or it requires manual cleanup? Maybe add a comment about that if unfortunately manual cleanup is required.

Reply:
It will not auto-recover, we will have to delete the configurations manually.
I will add the comment as below.
// Manual clean up of OVS configurations is required, when agent exits/aborts abruptly/abnormally.